### PR TITLE
out_cloudwatch_logs: increase MAX_EVENT_LEN to 1MB with tests

### PR DESCRIPTION
**Pull request by ShelbyZ**
**Summary**

CloudWatch logs [PutLogEvents API](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html) change in 2025 to support events up to 1MB ([ref](https://aws.amazon.com/about-aws/whats-new/2025/04/amazon-cloudwatch-logs-increases-log-event-size-1-mb/)) while our defaults reflect the older value 256KB.

Add runtime tests to validate the new limit:
- event_size_near_limit: Validates events just under 1MB are accepted
- event_size_at_aws_max: Validates events near AWS max are truncated
- event_truncation_with_backslash: Validates backslash handling at truncation boundary

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [N/A] Example configuration file for the change
- [X] Debug log output from testing the change

Gist: https://gist.github.com/ShelbyZ/76bfac405e03723efeb532a48d5b6a87

Over-limit on current logs
```
2025-11-13T04:37:45.155000+00:00 fluent-bit/log-router/b5075d48a899473fb8e8acc51e786ac2 [2025/11/13 04:37:45.155344086] [ warn] [output:cloudwatch_logs:app-logs-output] [size=1047894] Truncating event which is larger than max size allowed by CloudWatch
2025-11-13T04:37:45.155000+00:00 fluent-bit/log-router/b5075d48a899473fb8e8acc51e786ac2 [2025/11/13 04:37:45.155470414] [debug] [output:cloudwatch_logs:app-logs-output] Using stream=b5075d48a899473fb8e8acc51e786ac2, group=/ecs/fluentbit-workshop-lab3/app-logs
2025-11-13T04:37:45.159000+00:00 fluent-bit/log-router/b5075d48a899473fb8e8acc51e786ac2 [2025/11/13 04:37:45.155790284] [debug] [output:cloudwatch_logs:app-logs-output] cloudwatch:PutLogEvents: events=1, payload=262280 bytes
```

New limit
```
2025-11-13T04:32:20.986000+00:00 fluent-bit/log-router/21e880b6b1774c8e8ec2a6fb6b4679ea [2025/11/13 04:32:20.986187322] [debug] [output:cloudwatch_logs:app-logs-output] Using stream=21e880b6b1774c8e8ec2a6fb6b4679ea, group=/ecs/fluentbit-workshop-lab3/app-logs
2025-11-13T04:32:20.986000+00:00 fluent-bit/log-router/21e880b6b1774c8e8ec2a6fb6b4679ea [2025/11/13 04:32:20.986606024] [debug] [output:cloudwatch_logs:app-logs-output] cloudwatch:PutLogEvents: events=1, payload=1048057 bytes
2025-11-13T04:32:20.986000+00:00 fluent-bit/log-router/21e880b6b1774c8e8ec2a6fb6b4679ea [2025/11/13 04:32:20.986614394] [debug] [output:cloudwatch_logs:app-logs-output] Sending log events to log stream 21e880b6b1774c8e8ec2a6fb6b4679ea
```

Tested directly against CloudWatch Logs via PutLogEvents to discover that a maximum value of `1,048,546` bytes could be sent without issue. Using 1,000,000 bytes (1MB) gives a safe margin to avoid calculations for the `PUT_LOG_EVENTS_PAYLOAD_SIZE` which would need reevaluation to support observed max. The encoded max size was also `1,048,586` bytes which is 10 bytes about 1MiB (1,048,576) which makes me worry that depending on an exact max may break over time.

- [WIP] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [N/A] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [N/A] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [N/A] Documentation required for this feature

**Backporting**

- [N/A] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

## Summary by CodeRabbit

* **Improvements**
  * Increased maximum CloudWatch log event payload size from ~256 KB to 1 MB to permit larger single events.
  * Added a new public constant to control entity prefix length.

* **Tests**
  * Added end-to-end tests covering events at the size limit, events over the new limit, and truncation edge cases (including backslash handling).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: plugins/out_cloudwatch_logs/cloudwatch_api.h [hunk 1]
```diff
@@ -43,8 +43,8 @@
 /* Maximum number of character limits including both the Attributes key and its value */
 #define ATTRIBUTES_MAX_LEN 300
 
-/* 256KiB minus 26 bytes for the event */
-#define MAX_EVENT_LEN      262118
+/* 1MiB minus 26 bytes for the event */
+#define MAX_EVENT_LEN      1048550
```
[the remark this answers is not in the document]
Comment (ShelbyZ) [reply plugins/out_cloudwatch_logs/cloudwatch_api.h]: Working on the adjustment to the max size in a follow-up

[the remark this answers is not in the document]
Comment (ShelbyZ) [reply plugins/out_cloudwatch_logs/cloudwatch_api.h]: Resolved to use 1MB to avoid reevaluation calculations for `PUT_LOG_EVENTS_PAYLOAD_SIZE`

File: plugins/out_cloudwatch_logs/cloudwatch_api.h [hunk 2]
```diff
@@ -43,8 +43,17 @@
...
+/*
+ * https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
+ * AWS CloudWatch's documented maximum event size is 1,048,576 bytes (1 MiB),
+ * including JSON encoding overhead (structure, escaping, etc.).
+ * 
+ * Setting MAX_EVENT_LEN to 1,000,000 bytes (1 MB) provides a ~4.6% safety margin
+ * to account for JSON encoding overhead and ensure reliable operation.
+ * Testing confirmed messages up to 1,048,546 bytes (encoding to 1,048,586 bytes)
+ * succeed, though we use a conservative limit for production safety.
+ */
```
[the remark this answers is not in the document]
Comment (ShelbyZ) [reply plugins/out_cloudwatch_logs/cloudwatch_api.h:56]: Not useful, as this was the observed max values directly through API calls. I could remove the encoding to value but this was the value sent and successfully processed by CloudWatch.

[the remark this answers is not in the document]
Comment (ShelbyZ) [reply plugins/out_cloudwatch_logs/cloudwatch_api.h:56]: We can keep it as is

**Review by sparrc — APPROVED** [review]

**Comment by edsiper** [comment]
@codex review
